src/pybind/mgr/dashboard/controllers/user.py

```diff
@@ -89,3 +89,24 @@ def set(self, username, password=None, name=None, email=None, roles=None):
         user.set_roles(user_roles)
         mgr.ACCESS_CTRL_DB.save()
         return User._user_to_dict(user)
+
+
+@ApiController('/user/{username}')
+class UserChangePassword(BaseController):
+    @Endpoint('POST')
+    def change_password(self, username, old_password, new_password):
+        session_username = JwtManager.get_username()
+        if username != session_username:
+            raise DashboardException(msg='Invalid user context',
+                                     code='invalid_user_context',
+                                     component='user')
+        try:
+            user = mgr.ACCESS_CTRL_DB.get_user(session_username)
+        except UserDoesNotExist:
+            raise cherrypy.HTTPError(404)
+        if not user.compare_password(old_password):
+            raise DashboardException(msg='Old password do not match',
+                                     code='old_password_do_not_match',
+                                     component='user')
+        user.set_password(new_password)
```

Comment:
Rather than using a method here, Python provides `@property` for dealing with getter-setter logic. However I understand in this case with the current design it's not straight-forward, as we are mixing here: 1 controller + 1 thin model `User` (why in `services/access_control`?) + 2 different adapters (`JWTManager` + `ACCESS_CTRL_DB`):

```python
@property
def password(self):
  return self.__password

@password.setter
def password(self, value):
  if not password_valid(value):
    raise WhateverEx
  self.__password = value
  self.refreshLastUpdate()
  self.save() # if we don't want to perform per-field DB saves, we may force this happen externally
```

Reply:
I've choosen NOT to use ``@property`` here because other setters in this class are also not implemented this way.

Reply:
No problem.

Reply:
I'd suggest to submit a cleanup issue on our tracker to take care of this.